server/bleep/src/webserver/answer.rs

```diff
@@ -442,7 +438,16 @@ async fn handle_inner(
                     Stage::new("semantic_results", &all_snippets).with_time(stop_watch.lap()),
                 );
 
-                let filtered_snippets = deduplicate_snippets(all_snippets);
+                let query_embedding = semantic.embed(rephrased_query).map_err(|e| {
+                    error!("failed to embed query: {}", e);
+                    Error::internal(e)
+                })?;
+                let filtered_snippets = deduplicate_snippets(all_snippets, query_embedding);
+
+                event.write().await.stages.push(
```

Comment:
moved the analytics event up here, so we can see if `filtered_snippets` is empty in the pipeline.